### PR TITLE
Add callback entrypoints for per-frame Promise entrypoints.

### PR DESCRIPTION
**Pull request by kdashg**
Callbacks generate no new garbage themselves.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Mar 23, 2020, 9:27 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [HTML Diff Service](http://services.w3.org/htmldiff) - The HTML Diff Service is used to create HTML diffs of the spec changes suggested in a pull request.

:link: [Related URL](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fpr-preview.s3.amazonaws.com%2Fgpuweb%2Fgpuweb%2Fpull%2F626%2F2e86cff.html&doc2=https%3A%2F%2Fpr-preview.s3.amazonaws.com%2Fjdashg%2Fgpuweb%2Fpull%2F626.html)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 sysreq@w3.org to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20gpuweb/gpuweb%23626.)._
</details>

### CODE REVIEW

**Comment by Kangz** [comment]
Don't you need to make a closure or bind the callbacks to an object though?

**Comment by kainino0x** [comment]
Not if you need to capture the same things in the closure every time, which I think is usually possible. (e.g. bind to your engine's toplevel object once, save that bound function)

File: spec/index.bs [hunk 1]
```diff
@@ -1017,8 +1017,12 @@ its mapping.
      mapping.
 
 <script type=idl>
+callback BufferMapCallback = void (ArrayBuffer? mapping);
```
Comment (kvark) [inline spec/index.bs]: I think the user would want to have more information here, let's say the `GPUBuffer` as another argument?

**Comment by kainino0x** [comment]
cc #52 #214

**Comment by kainino0x** [comment]
Closing stale PR, I think the ship has sailed on this.

**Pull request closed without merging**